src/modules/scores.js

```diff
@@ -9,4 +17,83 @@ const refreshScores = () => {
   window.location.reload();
 };
 
-refreshButton.addEventListener('click', refreshScores);
\ No newline at end of file
+// Add new score to the list and save them into localStorage
+const addScore = () => {
+  const nameInput = document.querySelector('#name').value;
+  const scoreInput = document.querySelector('#score').value;
+
+  if (nameInput && scoreInput && scoreInput !== 0 && scoreInput > 0) {
+    // Save the data into the API
+    const myHeaders = new Headers();
+    myHeaders.append('Content-Type', 'application/json');
+
+    const raw = JSON.stringify({
+      user: nameInput,
+      score: parseInt(scoreInput, 10),
+    });
+
+    const requestOptions = {
+      method: 'POST',
+      headers: myHeaders,
+      body: raw,
+      redirect: 'follow',
+    };
+
+    fetch(`${urlAPI}${idAPI}/scores/`, requestOptions)
+      .then((response) => response.text())
+      .catch((error) => error);
+
+    // Reset the form and reload the page
+    setTimeout(() => {
+      document.querySelector('.add-form').reset();
+      window.location.reload();
+    }, 2000);
+  } else if (scoreInput === 0 || scoreInput < 0) {
+    errorElement.innerHTML = '<p>Score must be greater than 0</p>';
+    setTimeout(() => {
+      errorElement.innerHTML = '';
+    }, 3000);
+  } else {
+    errorElement.innerHTML = '<p>Please fill all the fields</p>';
+    setTimeout(() => {
+      errorElement.innerHTML = '';
+    }, 3000);
+  }
+};
+
+addButton.addEventListener('click', addScore);
```

Comment:
- [Optional] When the addButton is clicked, to make a post to the POST API, the score list does not need to be refreshed at this point as in your case. The GET API endpoint is only triggered when the refresh button is clicked to display the score on the board. You can equally consider fixing this :+1:

Reply:
Hi @FrederickMih
Thank you very much for your feedback!
I'm not sure if I understood this comment correctly. 
If you mean the `refreshButton.addEventListener('click', refreshScores);` they are that way because first declare the functions, and then create the `addEventListener` that invoke them. 
So the functions stayed together at the top and the listener together at the bottom.